docs/advanced/verifying-binaries.md

```diff
@@ -1,53 +1,305 @@
+an attacker could still potentially intercept and replace your download with
+their own malware. If you run the binary without verifying that it isn’t
+malware, you expose yourself to security risks.
+
+### How do you verify that a binary hasn’t been tampered with?
+
+The core concept of [file verification](https://en.wikipedia.org/wiki/File_verification)
+relies on hashing. The Decred team takes it a step further and uses
+[GnuPG](https://gnupg.org/index.html) to sign their hashes so you can verify
+that the file came directly from the Decred team.
+
+When the Decred team [compiles source code into binaries](https://github.com/decred/decred-release),
+they hash each binary using [SHA-256](https://en.wikipedia.org/wiki/SHA-2).
+Each hash is stored in a `manifest.txt` file. Then they sign the manifest using
+their public GnuPG key. This creates a signature file, ending in
+`manifest.txt.asc`.
+
+To check if your downloaded binary is a 100% match to the official Decred
+binary, you need to do two things:
+
+1. **Verify that the manifest signature was directly signed by Decred:**
+   Download the `manifest.txt` and the signature `manifest.txt.asc`. Use GnuPG
+   to check the signature against the public Decred Release Signing Key.
+
+     * SUCCESS: If the signatures match, you know you can trust that the
+       manifest came directly from Decred, and that the hashes inside it are
+       trustworthy.
+
+     * FAIL: If the signatures don't match, then your manifest is not the
+       official Decred manifest. Do not continue, delete the files you've
+       downloaded, and check for security problems before trying again.
+
+2. **Verify that the binary hash matches the manifest hash:** Hash your
+   downloaded binary and verify that its hash matches the hash in the official
+   Decred `manifest.txt`.
+
+     * SUCCESS: If your hash matches the hash in the manifest, you know you can
+       trust and run the binary.
+
+     * FAIL: If your hash doesn't match, your binary doesn't match the official
+       Decred binary release, and you should not continue. Delete the binary
+       you've downloaded, and check for security problems before trying again.
+
+## Detailed Instructions
+
+### Preliminary Steps
+
+First, go to the official [Decred releases](https://github.com/decred/decred-release/releases)
+page and download:
+
+* The binary installer for your OS / architecture
+* The file manifest and hashes, ending in `-manifest.txt`
+* The signature for the manifest, ending in `-manifest.txt.asc`
+
+Also, make sure GnuPG is installed:
+
+* On Windows, install GnuPG with the latest version of [gpg4win](https://www.gpg4win.org/).
+* On macOS, install GnuPG with the latest version of [GPGTools](https://gpgtools.org/).
+* On Linux, GnuPG is installed by default.
+
+With GnuPG installed, the following instructions will work from any terminal on
+Windows, macOS, or Linux/UNIX
+
+### Add the official Decred Release public keys to your GnuPG keyring.
```

Comment:
In all the cases below where I use "keys" instead of "key", it's because I'm referencing the entire keypair set. I just don't necessarily want to explain that in this guide.

When I import the Decred Release signing keys, and then run `gpg --list-keys`, it shows:
```------------------------------------------------
pub   rsa4096 2016-01-27 [SC]
      FD13B6835E248FAF4BD1838D6DF634AA7608AF04
uid           [ unknown] Decred Release <release@decred.org>
sub   rsa4096 2016-01-27 [S]
sub   rsa4096 2016-01-27 [E]
```

* `pub` [SC] is the primary public key which can be used for Signing/Certification.
* `sub` [S] is a public subkey which can be used for Signing.
* `sub` [E] is a public subkey which can be used for Encrypting.

So when you import the Decred Release signing keys, you get all three of these. However, in reality, you're only using the Signing Subkey (as referenced in #1046)

I think it's more correct to use "keys" plural, but it's semantics.

Reply:
Good point, let's leave it as plural